### PR TITLE
Keystore enhancement - added list, lock and unlock feature

### PR DESCRIPTION
**Pull request by josefwijaya**
- List : allow user to list all the keystore under the designated folder
- lock : allow user to lock a keystore account
- unlock: allow user to unlock a keystore, given a correct passphrase

### CODE REVIEW

File: cmd/gtxkey/unlock.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"log"
+
+	"github.com/abiosoft/ishell"
+	"github.com/gallactic/gallactic/cmd"
+	"github.com/gallactic/gallactic/keystore"
+)
+
+func Unlock() func(c *ishell.Context) {
+	return func(c *ishell.Context) {
+		// open keystore
```
Comment (b00f) [inline cmd/gtxkey/unlock.go]: Technically here we get the *instance* of keystore. It's loaded before.

File: keystore/keystore.go [hunk 2]
```diff
@@ -72,12 +74,65 @@ func (ks *Keystore) New(auth, label string, valAddr bool) (*KeyData, error) {
...
+	// get the keydata struct
+	keyObject, err := ks.getKeyObject(indexOrAddress)
+	if err != nil {
+		return err
+	}
+
+	// do decrypt and return the data
+	key, err := key.DecryptKeyFile(keyObject.Filename, auth)
+	if err != nil {
+		return fmt.Errorf("Decryption process fail")
```
Comment (b00f) [inline keystore/keystore.go]: 1- For warning/Error/Success messages use helper methods. look [`cmd/utils.go`](https://github.com/gallactic/gallactic/blob/d0947abb91a59569d6a18e152f52be2ce32e7693/cmd/utils.go#L165)
2- Better to report error message to the user. Look [here](https://github.com/gallactic/gallactic/blob/d0947abb91a59569d6a18e152f52be2ce32e7693/cmd/gallactic/key/inspect.go#L37).

File: cmd/gtxkey/unlock.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+		ks := c.Get("keystore").(*keystore.Keystore)
+
+		input := cmd.PromptInput("Key in account number/address to unlock the account (press ENTER to cancel): ")
+		auth := cmd.PromptPassphrase("Key in passphrase to unlock account("+input+") :", true)
+
+		err := ks.Unlock(auth, input)
+		if err == nil {
+			fmt.Println("Successfully Unlocked account")
```
Comment (b00f) [inline cmd/gtxkey/unlock.go]: Same, Please use `cmd.PrintSuccessMsg("Successfully Unlocked account")`

File: cmd/gtxkey/unlock.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+		} else {
+			log.Fatal(err)
```
Comment (b00f) [inline cmd/gtxkey/unlock.go]: `cmd.PrintErrorMsg(err)`

Comment (josefwijaya) [reply cmd/gtxkey/unlock.go]: resolved in [commit](https://github.com/gallactic/gallactic/pull/76/commits/91e618e9ee06fb2418c8cdbc25f9e69f530b6c14)

File: cmd/gtxkey/unlock.go (hunk 3 above)
Comment (josefwijaya) [reply cmd/gtxkey/unlock.go]: resolved  in [commit](https://github.com/gallactic/gallactic/pull/76/commits/91e618e9ee06fb2418c8cdbc25f9e69f530b6c14)

File: keystore/keystore.go (hunk 2 above)
Comment (josefwijaya) [reply keystore/keystore.go]: resolved  in [commit](https://github.com/gallactic/gallactic/pull/76/commits/91e618e9ee06fb2418c8cdbc25f9e69f530b6c14)

File: cmd/gtxkey/unlock.go (hunk 1 above)
Comment (josefwijaya) [reply cmd/gtxkey/unlock.go]: resolved  in [commit](https://github.com/gallactic/gallactic/pull/76/commits/91e618e9ee06fb2418c8cdbc25f9e69f530b6c14)

**Comment by b00f** [comment]
Thanks @josefwijaya

**Pull request merged**